### PR TITLE
Client handler

### PR DESCRIPTION
**Pull request by robertfalconer**
Simple search and results pages, read from html file templates

### CODE REVIEW

File: web/web.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+func rootHandler(writer http.ResponseWriter, request *http.Request) {
+	parseTemplate("search", writer, request)
+}
+
+func resultsHandler(writer http.ResponseWriter, request *http.Request) {
+	parseTemplate("results", writer, request)
+}
+
+func parseTemplate(templateName string, writer http.ResponseWriter, request *http.Request) {
+	htmlTemplate, err := loadTemplate(templateName, writer)
```
Comment (tyndyll) [inline web/web.go]: One thing you might want to consider is either passing in the template name to the loadTemplate function, or using the templateName as the name you pass to template.New. This just removes a magic string

File: web/web.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	if err != nil {
+		http.Error(writer, err.Error(), http.StatusNotFound)
+		return
+	}
+	err = htmlTemplate.Execute(writer, "template")
```
Comment (tyndyll) [inline web/web.go]: "template" here is the data you're putting into the template. If you don't have anything you can just pass in nil

Comment (robertfalconer) [inline web/web.go]: I thought it was the name of the template that was created with `template.New("template")`.

File: web/web.go (hunk 1 above)
Comment (robertfalconer) [inline web/web.go]: What do you mean a magic string?

**Comment by robertfalconer** [comment]
I've updated this to use `template.ParseFiles` instead of reading the template file and parsing the string.

**Comment by tyndyll** [comment]
:+1:

**Pull request merged**